flyway-core/src/main/java/org/flywaydb/core/internal/util/scanner/classpath/ClassPathScanner.java

```diff
@@ -308,11 +308,16 @@ private ClassPathLocationScanner createLocationScanner(String protocol) {
         }
 
         if ("jar".equals(protocol)
-                || "war".equals(protocol)
+                || "war".equals(protocol) //Tomcat
                 || "zip".equals(protocol) //WebLogic
                 || "wsjar".equals(protocol) //WebSphere
                 ) {
-            JarFileClassPathLocationScanner locationScanner = new JarFileClassPathLocationScanner();
+            JarFileClassPathLocationScanner locationScanner;
+            if ("war".equals(protocol)) {
+                locationScanner = new JarFileClassPathLocationScanner("*/");
```

Comment:
Hi, I manually did test this patch against Tomcat 8.5.5 and 8.5.13 (old and new URL handling) with plain 4.1.2 and patched 4.1.2.
The patched 4.1.2 always works correctly in all four constellations.
Do you need anything else to merge this PR?